### ISSUE TITLE
[Bug]: Got allows a redirect to a UNIX socket

### ISSUE THREAD

**Issue by benjamin-mogensen** [electron/electron#34949]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

19.0.8

### What operating system are you using?

macOS

### Operating System Version

macOS Monterey

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

No vulnerabilities reported.

### Actual Behavior

Vulnerabilities when doing `npm audit` and in Github:
```
electron@19.0.6 requires got@^9.6.0 via @electron/get@1.14.1
electron-builder@23.1.0 requires got@^9.6.0 via a transitive dependency on package-json@6.5.0
```
I am unable to get rid of these even updating both Electron and Electron Builder to their latest versions

**Comment by ckerr** [comment]
@benjamin-mogensen thanks for reporting this.

This is something that deserves to get fixed but since the fix can't be done by a commit in the `electron/electron` repo we probably shouldn't track it here. Please file new issues in upstream in https://github.com/electron/get/issues and https://github.com/electron-userland/electron-builder/issues so that the fixes can get done there. :bowing_man:

FWIW, the version of `got` in electron's own package.json is `"got": "^11.8.5"` which according to https://github.com/advisories/GHSA-pfrx-2q88-qq97 is a fixed version :tada:

**Comment by benjamin-mogensen** [comment]
@ckerr thanks for pointing me in the right direction. I can see `electron-get` already had the issue created. I have created an  issue for `electron-builder`.

**Comment by Dave645** [comment]
Hello, please reopen!

This issue has been fixed in `@electron/get@^2.0.0`, see https://github.com/electron/get/releases/tag/v2.0.0
Here is the PR merge: https://github.com/electron/get/pull/225

All is left is for `electron/electron` to commit the update to the latest branch. Hopefully it can make it in the `21.X.0` electron version.

After this electron will have 0 vulnerabilities according to npm.
